### PR TITLE
Allow to configure log scope key

### PR DESCRIPTION
**Pull request by laurynasr**
Allow to configure the log scope key for correlation ID property. This allows using a different key, such as `ActivityId` instead of hardcoded `CorrelationId`.

The default is `CorrelationId` as previously; custom key can be set when composing the application with `AddCorrelate`:

``` csharp
services.AddCorrelate(o => o.LoggingScopeKey = "ActivityId");
```

### CODE REVIEW

**Comment by laurynasr** [comment]
@skwasjer is there something else I need to do to get this merged? I see that the automated workflows are waiting for something but I don't know if I should do anything about that

**Comment by skwasjer** [comment]
@laurynasr no, I just have not had time to look at your proposal yet, will probably do over the weekend. Thanks for the submission.

File: src/Correlate.DependencyInjection/IHttpClientBuilderExtensions.cs [hunk 1]
```diff
@@ -35,6 +35,14 @@ public static IHttpClientBuilder CorrelateRequests(this IHttpClientBuilder build
...
         }
 
+        builder.Services.AddOptions<CorrelationManagerOptions>()
+            .Configure((CorrelationManagerOptions cmo) =>
+            {
+                var options = new CorrelateClientOptions();
+                configureOptions(options);
+                cmo.LoggingScopeKey = options.LoggingScopeKey;
+            });
+
```
Comment (skwasjer) [inline src/Correlate.DependencyInjection/IHttpClientBuilderExtensions.cs:45]: There are some issues with this setup. In case someone registers two different HttpClients but with different configuration, then the last configuration wins. E.g.:

```
services
    .AddHttpClient<MyService1>()
    .CorrelateRequests(opts => opts.LoggingScopeKey = "ActivityId");

services
    .AddHttpClient<MyService2>()
    .CorrelateRequests(opts => opts.LoggingScopeKey = "CorrelationId");
```

With the above, it seems to me both clients will still use `CorrelationId` as log scope key. That could be resolved by using named options I suppose, but I also don't think configuring the log scope key belongs in the client registration at all anyway. Currently it is thus possible to set the log scope key at the top level and per client. This seems redundant, as you'd really only want to use one unique log scope key across the entire application.

As such, I believe it should only be configurable via an explicit call to `AddCorrelate()`. The fact that `AddCorrelate()` is called from within this registration method should not be relevant if one calls `AddCorrelate(opts => opts.LogScopeKey = "ActivityId")` before registering clients (due to the use of `services.TryAdd<>`).

This also removes the need to derive the client options from `CorrelationManagerOptions`.

File: src/Correlate.Core/CorrelationManager.cs [hunk 2]
```diff
@@ -205,12 +212,12 @@ private T Execute<T>(string? correlationId, Func<T> correlatedFunc, OnException?
...
 
-    private static bool HandlesException<T>(OnException? onException, CorrelationContext correlationContext, Exception ex, out T result)
+    private bool HandlesException<T>(OnException? onException, CorrelationContext correlationContext, Exception ex, out T result)
     {
-        if (!ex.Data.Contains(CorrelateConstants.CorrelationIdKey))
+        if (!ex.Data.Contains(_options.LoggingScopeKey))
         {
             // Because we're about to lose context scope, enrich exception with correlation id for reference by calling code.
-            ex.Data.Add(CorrelateConstants.CorrelationIdKey, correlationContext.CorrelationId);
+            ex.Data.Add(_options.LoggingScopeKey, correlationContext.CorrelationId);
```
Comment (skwasjer) [inline src/Correlate.Core/CorrelationManager.cs:220]: Revert this change as it has nothing to do with log scope itself, even though the comment hints the relation to it. It is exception metadata.

**Review by skwasjer — CHANGES_REQUESTED** [review]

File: src/Correlate.Core/CorrelationManager.cs (hunk 2 above)
Comment (laurynasr) [reply src/Correlate.Core/CorrelationManager.cs]: Reverted exception data to always contain `CorrelationId` property.

File: src/Correlate.Core/CorrelationManager.cs [hunk 3]
```diff
@@ -43,14 +48,16 @@ ILogger<CorrelationManager> logger
...
     (
         ICorrelationContextFactory correlationContextFactory,
         ICorrelationIdFactory correlationIdFactory,
         ICorrelationContextAccessor correlationContextAccessor,
         ILogger<CorrelationManager> logger,
-        DiagnosticListener diagnosticListener
-    ) : this(correlationContextFactory, correlationIdFactory, correlationContextAccessor, logger)
+        DiagnosticListener diagnosticListener,
+        IOptions<CorrelationManagerOptions> options
+    ) : this(correlationContextFactory, correlationIdFactory, correlationContextAccessor, logger, options)
```
[the remark this answers is not in the document]
Comment (laurynasr) [reply src/Correlate.Core/CorrelationManager.cs]: Added a new constructor with all parameters instead of changing the existing one.

**Review by skwasjer — COMMENTED** [review]
lgtm. The only thing stopping the merge is some missing coverage on the new `AddCorrelate()` registration extension.

**Review by skwasjer — APPROVED** [review]

**Comment by skwasjer** [comment]
Thanks!
